web-api/elasticsearch/efcms-case-deadline-mappings.js

```diff
@@ -4,7 +4,7 @@ module.exports = {
       type: 'text',
     },
     'caseDeadlineId.S': {
-      type: 'text',
+      type: 'keyword',
```

Comment:
Consider introducing `index: false` to all columns that you expect to _never_ appear in the query predicates, but only returned in the `_source` arrays of queries.  This should also speed up indexing.

Reply:
@kkoskelin quick question about using `index:false`...it looks like [`elasticsearch-index-settings.js`](https://github.com/ustaxcourt/ef-cms/blob/staging/web-api/elasticsearch/elasticsearch-index-settings.js#L56) sets `dynamic: false` on all indexes which should prevent indexing any fields not explicitly set in the mapping. Is there any advantage to introducing `index:false` vs removing these fields from the mapping?

Reply:
@mark-meyer I am far from an elasticsearch expert -- so please take this all with a few grains of salt.
I think we may need some columns specified here so that their values are stored, and can then be retrieves using the `_sources` array, even if they're not part of the actual search predicates.
This is partly why I suggested a sort of "audit":
For any given ES index:
1) compile a list of columns mentioned in the `_sources`
2) ...and any columns and in the query predicates
3) any columns that appear ONLY in `_sources` should probably have `index: false` unless we think there's a chance that column will appear in a future query predicate.
4) any columns that appear in the index-mappings, but NOT in the `_sources` nor query predicates should be removed from the index-mappings.

I think.  I wouldn't mind folks checking my work on the above logic.  😄

Reply:
@kkoskelin I think these mappings only effect which terms get put in the inverted index. By default, the entire document will be saved to `_source` regardless of what we put in the mappings unless we explicitly exclude them with something like:

 ```
_source: {
    'excludes': [
      'some_field'
    ]
  }
```

On the other hand, no fields will be included in the inverted index (i.e. be searchable) unless they are included in the mapping above. So if we put nothing in the mapping, we would have the complete document saved in `_source`, but no searchable fields. 

We can see this in action with the seed data when the app is running locally. The `efcms-user-mappings.js` does not include the `email` field, but requesting all users with `http://localhost:9200/efcms-user/_search` shows emails like `"petitionsclerk@example.com"` in the `_source` object of the returned results. However, we can't search by email, because it's not in the mappings (and hence not in the index). 

That's not to say I think an audit is a bad idea — it makes sense to only index the things you are searching for. I just think the mechanism of `index: false` isn't needed.

Reply:
@mark-meyer I'll take your word for it -- so much of what I've learned is by reading and re-reading documentation and wishing we had an ES expert at our disposal.